clients/src/main/java/org/apache/kafka/clients/admin/internals/AdminMetadataManager.java

```diff
@@ -280,6 +281,9 @@ public void updateFailed(Throwable exception) {
         if (exception instanceof AuthenticationException) {
```

Comment:
Please add a function `isFatalException` and add AuthenticationException, AuthorizationException, SecurityDisabledException, UnsupportedEndpointTypeException, UnsupportedForMessageFormatException, UnsupportedVersionException etc.

You can add this function in `RequestUtils` class.

Reply:
Thanks for the review. This Jira would like to focus on fatal exception for metadata call in Admin. IIRC, The `SecurityDisabledException` is used in create / delete ACLs and `UnsupportedForMessageFormatException` is used in produce request, so I don't add them. Also, in `UnsupportedVersionException`, it logs different message for bootstrap and controller node, so I don't encapsulate all exceptions in a utility function.

Reply:
I understand that this is increasing the scope of the JIRA. What I am trying to suggest here will help us prevent such problems in future for cases where a client retries on a fatal exception. If we have a utility method, we can consolidate the logic of what is considered fatal at one place and existing/future calls to server APIs can use that utility to handle exceptions. It will prevent bugs in future if new authors can use the utility method.

Reply:
Thanks for the comment. Yeah, it's good to have a utility function for it. I add it to `RequestUtils#isFatalException`.